lua/orgmode/org/hyperlinks/init.lua

```diff
@@ -174,16 +174,32 @@ end
 ---@param headline OrgHeadline
 ---@param path? string
 function Hyperlinks.get_link_to_headline(headline, path)
-  path = path or utils.current_file_path()
   local title = headline:get_title()
-  local id
+
   if config.org_id_link_to_org_use_id then
-    id = headline:id_get_or_create()
+    local id = headline:id_get_or_create()
+    if id then
+      return ('id:%s::*%s'):format(id, title)
+    end
   end
 
-  if config.org_id_link_to_org_use_id and id then
-    return ('id:%s::*%s'):format(id, title)
+  path = path or utils.current_file_path()
+  return ('file:%s::*%s'):format(path, title)
+end
+
+---@param file OrgFile
+---@param path? string
+function Hyperlinks.get_link_to_file(file, path)
```

Comment:
Is a file path with the title valid format for a file link? 

From the examples here https://orgmode.org/guide/Hyperlinks.html I don't see any of those.

I think for link to file it is enough to either be an id, or just `file:/path/to/file.org` if there is no id.

Reply:
It is not, and it's also not treated such. I am piggy-backing here on the "title" format to sneak a description into insert_link to get it prefilled.
This is something, which actually needs to get cleaned up a bit more properly:
`get_link` should actually return even better an api equivalent of `Url`. And `insert_link` should take this. But this would need some more decent refactoring effort which touches a lot of places (think also of `store_link`) so I would prefer to defer this to a separate PR.

Right now I am quite happy, how Telescope-orgmode works although the API is not perfect yet.

Reply:
Ok, we can leave it for the time being, but later once we do some refactoring, I'd prefer to remove it.
It can also cause side effects in cases where `#+TITLE` or the file name is the same as some other headline value. In that case, the link would jump to the headline.